File: qualify-backend/pkg/certification.go
package pkg

type CertificationCreateRequest struct {
	Name        string `json:"name"`
	Year        int    `json:"year"`
	Description string `json:"description"`
	Institution string `json:"institution"`
}

type Certification struct {
	CertificationCreateRequest
	Id int `json:"id"`
}

type CertificationUpdateRequest struct {
	Name        *string `json:"name,omitempty"`
	Year        *int    `json:"year,omitempty"`
	Description *string `json:"description,omitempty"`
	Institution *string `json:"institution,omitempty"`
}

type CertificationsResponse struct {
	Certifications []Certification `json:"certifications"`
	Count          int             `json:"count"`
}

type CertificationResponse struct {
	Certification Certification `json:"certification"`
}

File: qualify-frontend/types/services/certification.ts
export interface Certification {
  id: number;
  name: string;
  description: string;
  institution: string;
  year: number;
}

export interface CertificationResponse {
  certification: Certification;
}

export interface CertificationsResponse {
  certifications: Certification[];
  count: number;
}

export interface CertificationUpdateRequest {
  name?: string;
  description?: string;
  institution?: string;
  year?: number;
}
